### PR TITLE
Convert ButtonLink into a styled component

### PR DESCRIPTION
**Pull request by brendonboshell**
# Description

In apps where we use react-router, we want a Trustyle button link that actually uses Link from react-router. We can use a styled component to achieve this, using it in this way:

```
<ButtonLink
      as={Link}
      to={href}
      onClick={onPlanInfoClick}
      variant="secondary"
    >
      Plan info
    </ButtonLink>
```

# Checklist

Pull request contains:

- [ ] A new component
- [X] Component maintenance: improvement / bug fix / etc
- [ ] Component library change: storybook / webpack / etc

This is a:

- [ ] A new component
- [ ] Small non-breaking change: patch version
- [X] Larger non-breaking change: minor version
- [ ] Breaking change: major version

Definition of done:

- [X] Work has been tested in multiple browsers
- [X] PR description includes description of change
- [X] If introducing a new component behaviour, added a story to cover that case.

### CODE REVIEW

File: src/elements/button-link/src/index.tsx [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
-      }}
-      href={href}
-      {...props}
-    >
-      {children}
-    </a>
-  )
-}
+}: Props<T>) => (
+  <Styled.a
```
Comment (callumacrae) [inline src/elements/button-link/src/index.tsx]: In my opinion, if this looks like a button, it should be `Styled.button`, not `Styled.a` - adding a `button` property to the `styles` object in the theme should make it available. Some of the styles below could be moved to it, as well

File: src/elements/button-link/src/stories.tsx [hunk 2]
```diff
@@ -38,3 +38,17 @@ export const PrimaryVariant = () => (
 export const SecondaryVariant = () => (
   <ButtonLink variant="secondary">Primary link button</ButtonLink>
 )
+
+const CustomLink: React.FC<React.AnchorHTMLAttributes<HTMLAnchorElement> & {
+  to: string
+}> = ({ to, children, ...rest }) => (
+  <a onClick={() => window.alert(`Custom link component: ${to}`)} {...rest}>
```
Comment (callumacrae) [inline src/elements/button-link/src/stories.tsx]: would be better to use a storybook action than an alert if it's available here

File: src/elements/button-link/src/index.tsx (hunk 1 above)
Comment (brendonboshell) [reply src/elements/button-link/src/index.tsx]: I thought that, but theme-ui's `Styled` doesn't have a button.

```
src/elements/button-link/src/index.tsx:25:11 - error TS2339: Property 'button' does not exist on type '{ p: SxComponent<ClassAttributes<HTMLParagraphElement> & HTMLAttributes<HTMLParagraphElement> & SxProps>; ... 27 more ...; root: SxComponent<...>; } & SxComponent<...>'.
```

Is it supposed to have it?

Comment (callumacrae) [reply src/elements/button-link/src/index.tsx]: Oh, that's weird! It's specifically mentioned in the docs as not available, but it doesn't explain why: https://theme-ui.com/theme-spec#styles

Carry on, then 😅

**Pull request merged**